cpp/generator.cpp

```diff
@@ -173,9 +174,11 @@ int64_t sample(const std::vector<float> &logits, double temperature) {
 
   int64_t sampled_token = distribution(gen);
   assert(0 <= sampled_token && sampled_token < VOCAB_SIZE);
-  return sampled_token;
+  double sampled_logit = logits[sampled_token];
+  return {sampled_token, sampled_logit};
```

Comment:
Logits are different from log-likelihood. Logits are raw outputs of the neural network, and they can be arbitrary numbers. Log-likelihood must be negative. In PyTorch, you can calculate log-likelihood from logits using https://pytorch.org/docs/stable/generated/torch.nn.functional.log_softmax.html#torch.nn.functional.log_softmax. Here you have to implement it in C++. Note that doing it naively may have numerical stability issues, but there is a simple trick (check PyTorch's implementation or ask me).

Reply:
Added translation trick and fixed! Thanks!

#### Current demo:
![image](https://github.com/lean-dojo/LeanInfer/assets/114432581/006cd673-9138-4295-aa50-a8bad2e358e0)